### PR TITLE
Support volumes propagation flags (Issue 554)

### PR DESCRIPTION
**Pull request by Hendrik-H**
The code change is based on the code for issue 688, which is about the same code area.

---
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/705)

### CODE REVIEW

File: src/main/java/com/github/dockerjava/api/model/PropagationMode.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+/**
+ * The propagation mode of a file system or file: <code>shared</code>, <code>slave</code> or <code>private</code>.
+ *
+ * @see https://github.com/docker/docker/pull/17034
+ * @since 1.22
+ */
+public enum PropagationMode {
+    /** default */
+    mount_default(""),
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/PropagationMode.java]: Name as java convention and annotate with JsonProperty according to real field.

Comment (Hendrik-H) [inline src/main/java/com/github/dockerjava/api/model/PropagationMode.java]: Problem is that the values are default and private, which is not allowed. So how should they be called then?
Do you have an example for the JsonProperty stuff? I tried to make this analogue to SELContext.java.

Comment (Hendrik-H) [inline src/main/java/com/github/dockerjava/api/model/PropagationMode.java]: I renamed the values but could not figure out where the JsonProperty annotation should go here. My code should be in line with `AccessMode` and `SELContext`.

Comment (Hendrik-H) [inline src/main/java/com/github/dockerjava/api/model/PropagationMode.java]: Please let me know if anything is missing. If something about Json is required then please let me know what exactly, ideally with a pointer to other similar code. Would be great if a version with this change could be released soon.

**Comment by KostyaSha** [comment]
LGTM but will back on weekend, sorry.

**Comment by KostyaSha** [comment]
#696 is the same options right? Could you review each other PR?

**Comment by Hendrik-H** [comment]
I had based my change on the branch for the other fix. So my branch contains the the same commit.
The other change looked good to me.

**Comment by dmandalidis** [comment]
@Hendrik-H please cherry-pick *again* that initial commit from #696 as it contains an integration test which is missing in your case. 

@KostyaSha this PR would be more complete in terms of features. However, I believe you could have pointed out to both of us that we 're doing the same thing in parallel while you were reviewing both PRs at the same time

**Comment by Hendrik-H** [comment]
Hi, I can only look into it on Monday. From what I had seen so far we added different things in the same area. Can't currently check the code. I thought one could first pull in the other change and then mine as I had based mine in the earlier commits.

**Comment by Hendrik-H** [comment]
I rebased my change now on top of the latest version of https://github.com/dmandalidis/docker-java/tree/issue-688. I'm still a bit new to git, so hope to have done it right ...

**Comment by dmandalidis** [comment]
Seems ok, please fixup your latest commit with the one that introduced the mistake.

**Comment by Hendrik-H** [comment]
sorry, no idea how I would do that. looks like I had the wrong version in 5abe479. so would somehow have to redo the whole thing. what's the problem of an additional commit ?

**Comment by dmandalidis** [comment]
https://github.com/docker-java/docker-java/commit/5abe4791c65681f9f7acca50432327e660c76383 introduced the wrong version. PR would be more clear that way, but again it's just my opinion :) See manual of git-rebase (along with the interactive option)

**Comment by Hendrik-H** [comment]
ok, I did a `git rebase --interactive 5abe479`, changed the order of the commits and changed the version fix from a `pick` to a `fixup`.

**Comment by Hendrik-H** [comment]
@KostyaSha anything missing to get this merged?

**Pull request merged**